### DISCUSSION TITLE
There was an error generating the provisioning profile ?

### DISCUSSION THREAD

**Discussion by martres** [fastlane/fastlane#19573] [Q&A]
Hello ! 
If someone can help, one of my certif expired, I just did a 
`fastlane match nuke`
and trying to make `fastlane match`

but i got this error 

`ERROR [2021-11-03 21:49:24.17]: Error finding relevant GitHub issues: undefined method `map' for nil:NilClass
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
Traceback (most recent call last):
	62: from /usr/local/bin/bundle:23:in `<main>'
	61: from /usr/local/bin/bundle:23:in `load'
	60: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/exe/bundle:37:in `<top (required)>'
	59: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	58: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/exe/bundle:49:in `block in <top (required)>'
	57: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:25:in `start'
	56: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	55: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:31:in `dispatch'
	54: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	53: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	52: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	51: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:478:in `exec'
	50: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:23:in `run'
	49: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:58:in `kernel_load'
	48: from /Library/Ruby/Gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:58:in `load'
	47: from /usr/local/bin/fastlane:23:in `<top (required)>'
	46: from /usr/local/bin/fastlane:23:in `load'
	45: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/bin/fastlane:23:in `<top (required)>'
	44: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	43: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	42: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	41: from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	40: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
	39: from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	38: from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	37: from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	36: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/commands_generator.rb:109:in `block (2 levels) in run'
	35: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	34: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	33: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	32: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	31: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	30: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	29: from Fastfile:28:in `block (2 levels) in parsing_binding'
	28: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	27: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
	26: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
	25: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	24: from Fastfile:79:in `block in parsing_binding'
	23: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	22: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	21: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	20: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	19: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	18: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	17: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	16: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/actions/sync_code_signing.rb:19:in `run'
	15: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/runner.rb:110:in `run'
	14: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/runner.rb:110:in `each'
	13: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/runner.rb:111:in `block in run'
	12: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/runner.rb:111:in `loop'
	11: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/runner.rb:112:in `block (2 levels) in run'
	10: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/runner.rb:269:in `fetch_provisioning_profile'
	 9: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/match/lib/match/generator.rb:109:in `generate_provisioning_profile'
	 8: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/sigh/lib/sigh/manager.rb:8:in `start'
	 7: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/sigh/lib/sigh/runner.rb:54:in `run'
	 6: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/sigh/lib/sigh/runner.rb:182:in `create_profile!'
	 5: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/spaceship/lib/spaceship/connect_api/models/profile.rb:76:in `create'
	 4: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/forwardable.rb:230:in `post_profiles'
	 3: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/spaceship/lib/spaceship/connect_api/provisioning/provisioning.rb:234:in `post_profiles'
	 2: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/spaceship/lib/spaceship/connect_api/provisioning/client.rb:37:in `post'
	 1: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/spaceship/lib/spaceship/connect_api/provisioning/client.rb:73:in `proxy_post'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/spaceship/lib/spaceship/connect_api/api_client.rb:194:in `handle_response': [!] { (Spaceship::UnexpectedResponse)
  "errors" : [ {
    "id" : "8ea79fdb-b7ba-4248-baf7-010343f26415",
    "status" : "409",
    "code" : "ENTITY_ERROR",
    "title" : "There is a problem with the request entity",
    "detail" : "There was an error generating the provisioning profile 'match PRODUCT_ID'.  Please try again.  If the problem persists, please contact Apple Developer Program Support. https://developer.apple.com/support"
  } ]
}
`

**Comment by lucgrabowski** [comment]
Hi, this is a response from Apple's backend. Do you try to use `PRODUCT_ID` variable in Matchfile? Could you show us Matchfile contents?

**Comment by martres** [comment]
Hello,  i am using `app_identifier`

This is my Matchfile

```
git_url("URL")
storage_mode("git")

type("appstore") # The default type, can be: appstore, adhoc, enterprise or development
app_identifier([ID1, ID2, ID3, ID4])
template_name("Associated Domains, Game Center, In-App Purchase, Push Notifications, Sign In with Apple")
```

**Comment by lucgrabowski** [comment]
You should check `template_name` description in https://docs.fastlane.tools/actions/match/ as it explains how to get `template_name` value from developer center. You put there things that are configured in app identifiers which causes issues on Apple's backend side.
